src/query/interpreter.cpp

```diff
@@ -145,28 +152,30 @@ class ReplQueryHandler final : public query::ReplicationQueryHandler {
 
   /// @throw QueryRuntimeException if an error ocurred.
   void SetReplicationRole(ReplicationQuery::ReplicationRole replication_role, std::optional<int64_t> port) override {
```

Comment:
Yes so atm we are sure because everything regarding replication is disabled (replication handler is disabled). So currently in my view not but if you really think we should, I can move check here also, I just didn't see any benefit from it at the moment of implementing